### ISSUE TITLE
`<Esc>` bug, and custom `<Esc>` mapping is removed

### ISSUE THREAD

**Issue by melkster** [gbprod/substitute.nvim#46]
This issue is two-fold. First of all, substitute.nvim removes custom `<Esc>` mappings when calling `exchange.operator()`. Secondly, I've noticed another bug related to `<Esc>`. Here's a config to reproduce the bug:

<details>
<summary><b>Click to expand</b></summary>

```lua
-- Ignore default config and plugins
vim.opt.runtimepath:remove(vim.fn.expand('~/.config/nvim'))
vim.opt.packpath:remove(vim.fn.expand('~/.local/share/nvim/site'))

-- Append test directory
local test_dir = vim.fn.expand('~/code-other/nvim-test-config')
vim.opt.runtimepath:append(vim.fn.expand(test_dir))
vim.opt.packpath:append(vim.fn.expand(test_dir))

-- Install packer
local install_path = test_dir .. '/pack/packer/start/packer.nvim'
local install_plugins = false

if vim.fn.empty(vim.fn.glob(install_path)) > 0 then
  vim.cmd('!git clone https://github.com/wbthomason/packer.nvim ' .. install_path)
  vim.cmd('packadd packer.nvim')
  install_plugins = true
end

local packer = require('packer')

packer.init({
  package_root = test_dir .. '/pack',
  compile_path = test_dir .. '/plugin/packer_compiled.lua'
})

packer.startup(function()
  function Use(module)
    use(require(string.format('configs.%s', module)))
  end

  -- Packer can manage itself
  packer.use 'wbthomason/packer.nvim'

  use { 'gbprod/substitute.nvim',
    config = function()
      local substitute = require('substitute')
      local exchange = require('substitute.exchange')

      vim.keymap.set('n', '<Esc>', function() print('escape') end)
      vim.keymap.set('n', 'sx', exchange.operator)

      substitute.setup()
    end
  }

  if install_plugins then
    packer.sync()
  else
    -- load plugins at your earliest convenience
    vim.defer_fn(function()
      vim.cmd('doautocmd User LoadPlugins')
    end, 1)
  end
end)
```

</details>

**To reproduce:**

1. `> nvim -u *path_to_config_above* *path_to_any_file*`
2. Press `sxiw` on some word
3. Press `<Esc>` twice
4. After the second `<Esc>`, insert mode is entered and the text ` 6>` (or similar) is inserted

I'm guessing that this is related to #43.

**Comment by gbprod** [comment]
Thanks for reporting this issue. I think that I see the problem. I try to fix it soon !

**Comment by gbprod** [comment]
Fixed by #47 
Can you confirm ?

**Comment by melkster** [comment]
@gbprod Works like a charm now! Thank you very much!
